sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -203,9 +203,26 @@ case class Cast(child: Expression, dataType: DataType, timeZoneId: Option[String
   // UDFToString
   private[this] def castToString(from: DataType): Any => Any = from match {
     case BinaryType => buildCast[Array[Byte]](_, UTF8String.fromBytes)
+    case StringType => buildCast[UTF8String](_, identity)
     case DateType => buildCast[Int](_, d => UTF8String.fromString(DateTimeUtils.dateToString(d)))
```

Comment:
we may covert a string to `UTF8String` and then convert it back, which is inefficient. I think we should create a special `StringBuilder` for `UTF8String`, e.g.
```
class UTF8StringBuilder {
  public void append(UTF8String str)
}
```

Reply:
I think this can also simplify the codegen part, as we have some duplicated code to avoid this conversion.

Reply:
yea, good idea. So, I'll add the builder class.